File: algorithms/php/20.ValidParentheses.php
<?php
class Solution {

    /**
     * @param String $s
     * @return Boolean
     */
    function isValid($s) {
        $stack = new SplStack();
        foreach (str_split($s) as $char) {
            if (in_array($char, ['{', '[', '('])) {
                $stack->push($char);
            }

            if (in_array($char, ['}', ']', ')'])) {
                if ($stack->isEmpty()) {
                    return false;
                }

                if (($char === '}' && $stack->top() === '{')
                    || ($char === ']' && $stack->top() === '[')
                    || ($char === ')' && $stack->top() === '(')) {
                    $stack->pop();
                } else {
                    return false;
                }
            }
        }

        return $stack->isEmpty();
    }
}

File: algorithms/cpp/20.ValidParentheses.cpp
#include "stack"
#include "functions.h"

using namespace std;

class Solution {
public:
    bool isValid(string s) {
        stack<char> stack;
        for (char ch : s) {
            if (ch == '{' || ch == '[' || ch == '(') {
                stack.push(ch);
            }

            if (ch == '}' || ch == ']' || ch == ')') {
                if (stack.empty()) {
                    return false;
                }

                if ((ch == '}' && stack.top() == '{')
                    || (ch == ']' && stack.top() == '[')
                    || (ch == ')' && stack.top() == '(')) {
                    stack.pop();
                } else {
                    return false;
                }
            }
        }

        return stack.empty();
    }
};

int main() {
    Solution s;
    cout << s.isValid("([)]");
}